cmake/extensions.cmake

```diff
@@ -20,17 +20,29 @@ function(get_board_without_ns_suffix board_in board_out)
   endif()
 endfunction()
 
-# Add a kconfig overlay file to a child image.
+# Append a value to a variable prefixed with a child image name.
+# This can be used by a parent image to modify the variables passed to its
+# child images. This must be invoked before 'add_child_image(image)'
+function(child_image_append_conf image val type)
+  set(old_conf ${${image}_${type}})
+  string(FIND "${old_conf}" "${val}" found)
+  if (${found} EQUAL -1)
```

Comment:
Will this work as expected ?
Imagine:
```
  child_image_append_conf(spm some_file.overlay DTC_OVERLAY_FILE)
```

then if user provides a custom setting, like:
`cmake .... -Dspm_DTC_OVERLAY_CONFIG="file1.overlay" <sample>`

then `old_conf` becomes: `file1.overlay` which will give `-1` when searching for this sting inside: `${val}` (== `some_file.overlay`) thus:
```
spm_OVERLAY_CONFIG="file1.overlay some_file.overlay"
```
now, if CMake is reinvoked, then `spm_OVERLAY_CONFIG` is now: `file1.overlay some_file.overlay`, but that string is nowhere to be found inside `${val}` (== `some_file.overlay`), thus after second cmake, we'll have:
```
spm_OVERLAY_CONFIG="file1.overlay some_file.overlay some_file.overlay"
```